pkg/engine/variables/operator/anynotin_test.go

```diff
@@ -0,0 +1,126 @@
+package operator
+
+import (
+	"testing"
+
+	"github.com/go-logr/logr"
+	"github.com/kyverno/kyverno/pkg/engine/context"
+)
+
+func TestAnyNotInHandler_Evaluate(t *testing.T) {
+	type fields struct {
+		ctx context.EvalInterface
+		log logr.Logger
+	}
+	type args struct {
+		key   interface{}
+		value interface{}
+	}
+	tests := []struct {
+		name   string
+		fields fields
+		args   args
+		want   bool
+	}{
+		{
+			name: "key is string and in value",
+			args: args{
+				key:   "kyverno",
+				value: "kyverno",
+			},
+			want: false,
+		},
+		{
+			name: "key is string and in value",
+			args: args{
+				key:   "kube-system",
+				value: []interface{}{"default", "kube-*"},
+			},
+			want: false,
+		},
+		{
+			name: "key is string and not in value",
+			args: args{
+				key:   "kyverno",
+				value: "default",
+			},
+			want: true,
+		},
+
+		{
+			name: "key is int and in value",
+			args: args{
+				key:   64,
+				value: "64",
+			},
+			want: false,
+		},
+		{
+			name: "key is int and not in value",
+			args: args{
+				key:   64,
+				value: "default",
+			},
+			want: true,
+		},
+		{
+			name: "key is array and in value",
+			args: args{
+				key:   []interface{}{"kube-system", "kube-public"},
+				value: "kube-*",
+			},
+			want: false,
+		},
+		{
+			name: "key is array and partially in value",
+			args: args{
+				key:   []interface{}{"kube-system", "default"},
+				value: "kube-system",
+			},
+			want: true,
+		},
+		{
+			name: "key is array and not in value",
+			args: args{
+				key:   []interface{}{"default", "kyverno"},
+				value: "kube-*",
+			},
+			want: true,
+		},
+		{
+			name: "key and value are array and not in value",
+			args: args{
+				key:   []interface{}{"default", "kyverno"},
+				value: []interface{}{"kube-*", "kube-system"},
+			},
+			want: true,
+		},
+		{
+			name: "key and value are array and partially in value",
+			args: args{
+				key:   []interface{}{"default", "kyverno"},
+				value: []interface{}{"kube-*", "ky*"},
+			},
+			want: true,
+		},
+		{
+			name: "key is empty but is in value",
+			args: args{
+				key:   []interface{}{},
+				value: []interface{}{"default", "kyverno"},
+			},
+			want: false,
```

Comment:
What does the empty key mean? An empty string?

Reply:
@Chandan-DK ?

Reply:
The empty key here refers to an empty array. I rephrased the name of the test for clarity.
I also wrote another test for an empty string as the key.

Reply:
Why does an empty array match `[]interface{}{"default", "kyverno"}`?

Reply:
@realshuting It is actually a case to be handled by the way the policy is written. 
Please check these discussions
https://github.com/kyverno/kyverno/issues/3569
https://kubernetes.slack.com/archives/CLGR9BJU9/p1649411903315429


Like @chipzoller said 
https://github.com/kyverno/kyverno/pull/7972#issuecomment-1670277564 https://github.com/kyverno/kyverno/pull/7972#issuecomment-1673045629 
the unit test I added can be considered as one of the scenarios

Reply:
I see what the point is. It makes sense to me for Any**Not**In.